async-usercalls/src/alloc/mod.rs

```diff
@@ -0,0 +1,69 @@
+use std::cell::RefCell;
```

Comment:
> this is increasing the number of different memory allocator implementations

I don't think that's necessarily a bad thing, unless you can think of a way to have a unified implementation for some/all of those.

> Couldn't you just do the equivalent this every you need to do a usercall?

I initially implemented it that way, however I should point out that swapping that implementation with this allocator + simplified usercalls did not result in much change in the number of lines of code. And the allocator is additionally useful for sync usercalls.

Although my main motivation for introducing the allocator was reducing the number of enclave exits per each usercall, I no longer think that would be a big factor since the only calls that benefit from this are bind/accept/connect which are not the most critical to optimize for (apart from maybe accept?). But I don't see any downside to it either.